From de727f6b99b2b095d3b0ad7c2464fbc5c4f0d42f Mon Sep 17 00:00:00 2001
From: Jerry Jelinek <jerry.jelinek@joyent.com>
Date: Mon, 14 Nov 2016 11:38:38 -0700
Subject: [PATCH] HEAD-2329 hook cnapi changes for RFD 53 into joysetup
 Reviewed by: Pedro Palazon Candel <pedro@joyent.com> Reviewed by: Josh
 Wilsdon <jwilsdon@joyent.com> Approved by: Pedro Palazon Candel
 <pedro@joyent.com>

---
 scripts/joysetup.sh | 17 +++++++++++++++--
 1 file changed, 15 insertions(+), 2 deletions(-)

diff --git a/scripts/joysetup.sh b/scripts/joysetup.sh
index 1c0d6b0c..e8dc8eda 100755
--- a/scripts/joysetup.sh
+++ b/scripts/joysetup.sh
@@ -6,7 +6,7 @@
 #
 
 #
-# Copyright (c) 2014, Joyent, Inc.
+# Copyright 2016, Joyent, Inc.
 #
 
 #
@@ -622,7 +622,20 @@ if [[ "$(zpool list)" == "no pools available" ]] \
         POOL_FILE=/mockcn/${MOCKCN_SERVER_UUID}/pool.json
     fi
 
-    if ! /usr/bin/disklayout "${arg_disklayout}" > ${POOL_FILE}; then
+    # We expect sdc-server to validate the input for the following parameters
+    # passed by CNAPI, but to be safe, we use some defensive quoting as well.
+    if [[ -n "${CONFIG_spares}" ]]; then
+        dl0="-s"
+        dl1="${CONFIG_spares}"
+    fi
+    if [[ -n "${CONFIG_cache}" && "${CONFIG_cache}" == "false" ]]; then
+        dl2="-c"
+    fi
+    if [[ -n "${CONFIG_layout}" ]]; then
+        dl3="${CONFIG_layout}"
+    fi
+
+    if ! /usr/bin/disklayout ${dl0} "${dl1}" ${dl2} "${dl3}"> ${POOL_FILE}; then
         fatal "disk layout failed"
     fi
 
-- 
2.21.0

